pipeline/pipeline.go

```diff
@@ -474,9 +514,34 @@ func (p *Pipeline) expandProcs() {
 	p.procCount.Swap(to)
 }
 
+// DeltaWrapper acts as a wrapper around int64
+// and returns the difference between
+// the new and the old value when a new value is set.
+// This is useful for metrics (see the maintenance function).
+type DeltaWrapper struct {
+	val int64
+}
+
+func newDeltaWrapper() *DeltaWrapper {
+	return &DeltaWrapper{0}
+}
+
+func (c *DeltaWrapper) setAndGetDiff(newVal int64) int64 {
```

Comment:
Weak naming, `DeltaWrapper` sets `val`, not `diff` which not obvious.
My suggestion:
```
// updateDelta sets new Delta and returns diff to previous Delta.
func (c *DeltaWrapper) updateDelta(val int64) (diff int64) {
	diff := newVal - c.val
	c.val = newVal
	return diff
}
```
Named return `diff` appropriate here, it describes returning int64.

And why receiver short name is `c`? `dw` seems better.

Reply:
done

Reply:
I decided to name the function `updateValue` and use `diff` as the name for the returned value.  
That way we know we change the value (not the delta) and we get a `diff`erence between the new value and the old one.